packages/integration/src/gitlab/GitLabIntegration.ts

```diff
@@ -62,6 +81,92 @@ export class GitLabIntegration implements ScmIntegration {
   resolveEditUrl(url: string): string {
     return replaceGitLabUrlType(url, 'edit');
   }
+
+  fetch(input: RequestInfo | URL, init?: RequestInit): Promise<Response> {
+    return this.fetchImpl(input, init);
+  }
+
+  private createFetchStrategy(): FetchFunction {
+    let fetchFn: FetchFunction = async (url, options) => {
+      return fetch(url, { ...options, mode: 'same-origin' });
+    };
+
+    // Early return for plain fetch (no enhancements needed)
+    if (
+      this.integrationConfig.limitPerMinute <= 0 &&
+      this.integrationConfig.maxRetries <= 0
+    ) {
+      return fetchFn;
+    }
+
+    // Apply retry wrapper if configured
+    if (this.integrationConfig.maxRetries > 0) {
+      fetchFn = this.withRetry(fetchFn);
+    }
+
+    // Apply throttling wrapper if configured
+    if (this.integrationConfig.limitPerMinute > 0) {
+      fetchFn = pThrottle({
+        limit: this.integrationConfig.limitPerMinute,
+        interval: 60_000,
+      })(fetchFn);
+    }
+
+    return fetchFn;
+  }
+
+  private withRetry(fetchFn: FetchFunction): FetchFunction {
+    return async (url, options) => {
+      let retries = 0;
+
+      while (retries <= this.integrationConfig.maxRetries) {
+        this.queue.push(() => {});
+
+        const response = await fetchFn(url, options);
```

Comment:
This seems to me that it needs a try/catch. And especially [for AbortError](https://developer.mozilla.org/en-US/docs/Web/API/AbortSignal/reason) it should be immediately short circuited and return